crypto/src/hash/rpo/digest.rs

```diff
@@ -0,0 +1,112 @@
+use super::DIGEST_SIZE;
+use core::slice;
+use winterfell::crypto::Digest;
+use winterfell::math::{fields::f64::BaseElement, StarkField};
+use winterfell::{ByteReader, ByteWriter, Deserializable, DeserializationError, Serializable};
```

Comment:
A couple of thoughts:

- We should probably use `Felt` instead of `BaseElement` everywhere (and it can be imported from `crate`).
- Now that I think about this, instead of importing full `winterfell` crate (in `Crates.io`) we could probably import just the relevant crates (e.g., `winter-math`, utils etc.).

Reply:
I did as you suggested with respect to the second point. The only issue I faced was when I tried to do it for winter_utils in lib.rs so that we can use import in the sub-modules. For some reason I get some weird errors related to trait bounds.